src/apps/replication/lib/mutation_log.cpp

```diff
@@ -1078,7 +1080,7 @@ int mutation_log::garbage_collection(replica_log_info_map& durable_decrees)
     else
     {
         // the last one should be the current log file
-        dassert(files.rbegin()->first == current_file_index, "");
+        dassert(-1 == current_file_index || files.rbegin()->first == current_file_index, "");
```

Comment:
Current_log_file is created on demand. when there is no incoming mutation from being started, and gc happens, the current_log_file will be empty.